### DISCUSSION TITLE
Prompt Database

### DISCUSSION THREAD

**Discussion by LilithDragoness** [lllyasviel/stable-diffusion-webui-forge#1796] [Ideas]
It would be incredibly useful to have a prompt database within Forge, where the prompt for each generated image is stored, by default, for up to three months.

Since the file names seem unique, they could be used as the primary key. Alternatively, the MD5 hash of the image could serve the same purpose.

This feature would be particularly valuable for batch processing, such as upscaling Flux-generated images using SDXL, especially when the images were generated with the "Prompts from file or textbox" script. Another use case would be fixing batch-generated images via inpainting.

For example, if a user generates 50 images using the textbox prompt script and 10 of those images need finger corrections with inpaint, it's often more effective to use the original prompt with Flux rather than a generic prompt like "hands" or "fingers."

A checkbox could be added next to the prompt field with the option: "Use the original prompt for this image." When selected, the current prompt would be appended to the original one.

**Comment by Juqowel** [comment]
Let's be clear first. Did you know that you can drag and drop a generated image into a prompt area?
Prompt(+settings) already saved into image itself.

**Reply by renega51** [reply]
I just learned what you said, but the prompts are displayed on the screen in Chinese. But the metadata is in English, and I created these images in English myself.

**Comment by LilithDragoness** [comment]
I didn’t know that because Fooocus didn’t include this feature. Thank you for letting me know.

I think it’s a terrible idea to have such a feature enabled by default. I upload my images to various platforms, and I absolutely do not want to share my prompts with anyone.

That said, I wonder if this could be useful for batch processing. Then I would remove the prompt info only from the final images.

For now, I’ll need to scan all my Forge images and remove the EXIF data.

**Reply by Juqowel** [reply]
> and I absolutely do not want to share my prompts with anyone.

I use ExifCleaner app for remove but you can find something similar.

**Reply by LilithDragoness** [reply]
> > and I absolutely do not want to share my prompts with anyone.
> 
> I use ExifCleaner app for remove but you can find something similar.

```exiftool -r -Parameters= -overwrite_original -ext jpg -ext gif -ext png -ext jpeg /path/to/images```

This will not remove the color profile and other information you might want to keep. I'm a bit sad because updating the images generates a lot of noise on my Google Drive, and I often rely on the creation and modification times of the files.

We should have am option in the Settings to turn this off.

**Comment by LilithDragoness** [comment]
I couldn’t even find an option in the settings to turn it off.

**Reply by DenOfEquity** [reply]
Settings -> (User Interface) Infotext -> Write infotext to metadata of the generated image

**Reply by LilithDragoness** [reply]
> Settings -> (User Interface) Infotext -> Write infotext to metadata of the generated image

![00206-8587](https://github.com/user-attachments/assets/99341d1e-74a6-4329-b279-62c94bcbf79a)

**Comment by LilithDragoness** [comment]
By the way, we haven’t discussed the concept I described, which can’t be resolved by any of the mentioned solutions. If I recall correctly, someone mentioned an extension that stores prompts, but I can’t find that comment anymore.

The key issue is that none of these options assist with batch processing. The database I proposed would address that

**Reply by mold0101** [reply]
Infinite image browsing

**Reply by LilithDragoness** [reply]
Earlier, someone mentioned sd-webui-prompt-history, but the comment has since been removed.

I’m wondering if anyone had a chance to read what I originally wrote.

I wasn’t referring to simply saving the prompt—that can be easily done with a text editor. Nor was I talking about reapplying it manually through actions like clicking, dragging, or dropping—that’s just as easy with copy-paste.

What I was referring to is automatically applying the prompt to the image, potentially as part of a batch-processing workflow, where each image might have a different prompt.
